.github/workflows/testing.yml

```diff
@@ -65,51 +67,30 @@ jobs:
-        name: Get datalad - Windows
-        run: pip install datalad==0.16.3
+          key: behavior-datasets-2022-08-18-${{ matrix.os }}-${{ steps.behavior.outputs.HASH_behavior_DATASET }}
+
 
-      - if: steps.cache-ecephys-datasets.outputs.cache-hit == false && (matrix.os == 'ubuntu-latest' || matrix.os == 'windows-latest')
-        name: "Force GIN: ecephys download"
+
+      - if: steps.cache-ephys-datasets.outputs.cache-hit == false || steps.cache-ophys-datasets.outputs.cache-hit == false || steps.cache-behavior-datasets.outputs.cache-hit == false
+        name: Install and configure AWS CLI
         run: |
-          datalad install https://gin.g-node.org/NeuralEnsemble/ephy_testing_data
-          cd ephy_testing_data
-          datalad get -r ./neuralynx/Cheetah_v5.4.0/original_data/
-          datalad get -r ./neuralynx/Cheetah_v5.5.1/original_data/
-          datalad get -r ./neuralynx/Cheetah_v5.6.3/original_data/
-          datalad get -r ./neuralynx/Cheetah_v5.7.4/original_data/
-          datalad get -r ./neuroscope/
-          datalad get -r ./openephysbinary/v0.4.4.1_with_video_tracking/
-          datalad get -r ./blackrock/
-          datalad get -r ./intan/
-          datalad get -r ./spikegadgets/
-          datalad get -r ./spike2/
-          datalad get -r ./spikeglx/Noise4Sam_g0/Noise4Sam_g0_imec0/
-          datalad get -r ./spikeglx/TEST_20210920_0_g0/
-          datalad get -r ./phy/phy_example_0/
-          datalad get -r ./axona/
-          datalad get -r ./cellexplorer/
-          datalad get -r ./axon/
-          datalad get -r ./edf/
-          cd ..
-      - if: steps.cache-ophys-datasets.outputs.cache-hit == false && (matrix.os == 'ubuntu-latest' || matrix.os == 'windows-latest')
-        name: "Force GIN: ophys download"
-        run: datalad install -rg https://gin.g-node.org/CatalystNeuro/ophys_testing_data
-      - if: steps.cache-behavior-datasets.outputs.cache-hit == false && (matrix.os == 'ubuntu-latest' || matrix.os == 'windows-latest')
-        name: "Force GIN: behavior download"
-        run: datalad install -rg https://gin.g-node.org/CatalystNeuro/behavior_testing_data
+          pip install awscli==1.25.27
+          aws configure set aws_access_key_id ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws configure set aws_secret_access_key ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+
+      - if: steps.cache-ephys-datasets.outputs.cache-hit == false
+        name: Download ephys dataset from S3
+        run: aws s3 cp --recursive ${{ secrets.S3_GIN_BUCKET }}/ephy_testing_data ./ephy_testing_data
+      - if: steps.cache-ophys-datasets.outputs.cache-hit == false
+        name: Download ophys dataset from S3
+        run: aws s3 cp --recursive ${{ secrets.S3_GIN_BUCKET }}/ophys_testing_data ./ophys_testing_data
```

Comment:
`secrets.S3_GIN_BUCKET`. Is this just the S3 URI? I am curios on why this need to be a secret as even if the URI is public you will not be able to access, right?

Reply:
Don't want to confuse people mainly - the public-facing avenues are the GIN repo's, and that's where individuals should try to get it from (either through GIN CLI or datalad), it's just that S3 is much easier for automated workflows in the cloud (which are essentially just containers).

Reply:
> > Any reason for using the cli utility here awscli and not the keithweaver/aws-s3-github-action@v1.0.0 as you do with the cron for caches?
> 
> Yep, reason given in [#102 (comment)](https://github.com/catalystneuro/neuroconv/pull/102#issuecomment-1221411627)

So you can't use the action in other platforms that are not linux. Why not use the cli everywhere then? (to make our action code more similar)

Reply:
Theoretically, haven't put that to the test

Reply:
I tried it this morning. You can't even look at it from the URI.